src/wasm/wasm-type.cpp

```diff
@@ -43,49 +43,46 @@ size_t hash<wasm::Type>::operator()(const wasm::Type& type) const {
 }
 
 size_t hash<wasm::Signature>::operator()(const wasm::Signature& sig) const {
-  auto res = hash<uint64_t>{}(sig.params.getID());
-  wasm::hash_combine(res, sig.results.getID());
-  return res;
+  return wasm::rehash(uint64_t(hash<uint64_t>{}(sig.params.getID())),
+                      uint64_t(hash<uint64_t>{}(sig.results.getID())));
 }
 
 size_t hash<wasm::Field>::operator()(const wasm::Field& field) const {
-  auto res = hash<uint64_t>{}(field.type.getID());
-  wasm::hash_combine(res, uint32_t(field.packedType));
-  wasm::hash_combine(res, field.mutable_);
-  return res;
+  return wasm::rehash(uint64_t(hash<uint64_t>{}(field.type.getID())),
+                      uint64_t(field.packedType) << 1 | field.mutable_);
 }
 
 size_t hash<wasm::TypeDef>::operator()(const wasm::TypeDef& typeDef) const {
   auto kind = typeDef.kind;
-  auto res = hash<uint32_t>{}(uint32_t(kind));
+  auto res = wasm::rehash(uint64_t(0), uint64_t(kind));
```

Comment:
From here on I am trying to always use `uint64_t` without switching to 32-bit from the start...